### DISCUSSION TITLE
Failed parsing SMILES

### DISCUSSION THREAD

**Discussion by hongxianglics** [rdkit/rdkit#5844] [General] [answered]
Hi,

I meet the problem that the SMILES can not be parsed.

I have a sample here:
below is the code I write:
<img width="397" alt="捕获" src="https://user-images.githubusercontent.com/104338096/206086513-dbca8cea-38a5-4b6b-8e7d-9d11a4c28e3c.PNG">

and I got this result

<img width="722" alt="捕获1" src="https://user-images.githubusercontent.com/104338096/206086521-73fa42ba-8114-4974-a44f-7c8d3b75f2ac.PNG">

I also attached a screenshot here. I hope the SMILES can be parsed correct.

Thanks!

**Comment by apahl** [comment, marked as answer]
Hi,
the pipe character ("|") in the middle of the SMILES string is not a valid character in a SMILES, therefore the parsing fails.
It seems that both parts of the string (to the left and to the right of the pipe character) are identical, so if you just use one half of the string, it should work:

```Python
mol = "NC[C@H](O)C(=O)O"
x = Chem.MolFromSmiles(x)
```

![image](https://user-images.githubusercontent.com/9406836/206131628-46941c76-cbe8-4180-886d-b291edde8117.png)


KR Axel

**Comment by hongxianglics** [comment]
I see! Thanks very much for your help!
